service/messagecontext/message_context.go

```diff
@@ -0,0 +1,39 @@
+// Package messagecontext stores and accesses the message struct in
+// context.Context.
+package messagecontext
```

Comment:
Using the context initializer we can use a communication context package to communicate between resources. This message context is the medium used to e.g. provide information from the config map resource to the deployment resource. I quite like using the context as communication channel.

Reply:
Not sure if you guys have already discussed about this, I'm not superfan of storing custom values on the context, IMO the parameters of functions should be clear, it is easier for everyone to reason about how things work. 

I'd like to see an example of when the message is updated, would the info put in it be request-scoped or would it carry some additional state?

Also, the logic is depending on the existence of these values, nothing would work if the messages are not set. Not sure if this is the intended usage of value in contexts (should be more informational IMO), WDYT?

Reply:
Pawel and I talked a bit about this. Using the context as communication channel is the best option we came up with. When you have a better idea please share it with us. The reconciler framework tries to be generic. The only specific argument resources get is the context. The other arguments are the custom object and the resource specific state. 

One alternative would be some service that two resources get configured with. This new dependency could provide read and write primitives. The problem with this is that we would need to manage something like global state which has to be namespaced and flushed with each reconciliation loop. The context's nature is to be volatile on a per "request" level, which is the reconciliation loop here, which is why we decided to go with this.